### DISCUSSION TITLE
Barcode of Order ID & Shipping Tracking Number

### DISCUSSION THREAD

**Discussion by davidhardik** [PrestaShop/PrestaShop#34704] [Ideas]
#### Is your feature request related to a problem?

I think it is frustrating not to have a barcode placed for shipping number and of Order ID in Delivery slip.

#### Describe the solution you'd like

A way to showcase barcode using the Shipping tracking number given in order and the order ID generated. It is helpful rather than looking for plugins and increasing the load time and processes. So, if it is possible to integrate then kindly do it as it is a common issue being faced majorly on many platforms without any solutions.

[a comment by an automated account is not shown]

**Comment by colinegin** [comment]
Hello @davidhardik ,

Thanks for your feedback.
So If try to sum up what you just said, you would like the shipping tracking number (+ ID of the order) to be transformed into a barcode that can be scanned ?
When would that code be scanned ? When the shipping is in progress ? 
What would be the action followed by this scan ? 

Thanks for you reply, this will help us have a better understanding of the feature that you would need.

**Comment by davidhardik** [comment]
Hi,

Yes delivery slips acts as Shipping Label so it will be used for scanning by Courier or Logistic Partners. Giving the Tracking number helps user track their information on courier partner website. It remains easier for them to receive the delivery slips with order ID & Tracking URL or number so that they can track it whenever they want.

The code will be scanned on the "Processing in Progress" & "Shipping in Progress" State or Dispatch Ready State or status.

Actually the action will be taken care by the Courier or logistic partners but a person who is managing the backend will be able to generate the delivery slips and pass it on to the Shipping or courier service and even the customers.

Sometimes merchant wants to display EAN13, UPC & others as barcode on invoices or delivery slips so providing an option for custom or inbuilt to create barcode of will certainly help.
![pdf_samples_simple2-3](https://user-images.githubusercontent.com/66987437/84877416-19ac3f80-b0a6-11ea-8d96-3d202c67fdf4.jpg)


It will be a helpful feature if in invoice we will be able to put tracking ID as barcode as well.

Let me know in case you need more details.
